File: teamarr/api/routes/support.py
"""Share-safe support bundle download endpoint."""

from datetime import datetime

from fastapi import APIRouter
from fastapi.concurrency import run_in_threadpool
from fastapi.responses import Response

from teamarr.services.support_bundle import SupportBundleService

router = APIRouter(prefix="/support")


@router.get("/bundle")
async def download_support_bundle() -> Response:
    """Download a redacted diagnostic archive without changing application state."""
    content = await run_in_threadpool(SupportBundleService().create)
    filename = datetime.now().strftime("teamarr-support-%Y%m%d-%H%M%S.zip")
    return Response(
        content=content,
        media_type="application/zip",
        headers={"Content-Disposition": f'attachment; filename="{filename}"'},
    )

File: frontend/src/api/support.ts
/** Download the share-safe support bundle without using the JSON API client. */
export async function downloadSupportBundle(): Promise<void> {
  const response = await fetch("/api/v1/support/bundle")
  if (!response.ok) {
    throw new Error("Failed to create support bundle")
  }

  const disposition = response.headers.get("Content-Disposition")
  const filename = disposition?.match(/filename="?([^";]+)"?/)?.[1] ?? "teamarr-support.zip"
  const url = URL.createObjectURL(await response.blob())
  const link = document.createElement("a")
  link.href = url
  link.download = filename
  link.click()
  URL.revokeObjectURL(url)
}
